bundles/org.openhab.core.config.discovery.usbserial.otheros/src/main/java/org/openhab/core/config/discovery/usbserial/otheros/internal/UsbProductDatabase.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.config.discovery.usbserial.otheros.internal;
+
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * This is a database map between hex 'vendorId:productId' pairs and the descriptions of known products. This is
+ * required because in some cases the USB scanners fail to read the product description strings, so this provides
+ * fallback descriptions.
+ * <p>
+ * It contains a map of vendor ids, product ids, and product descriptions. The descriptions should include at least a
+ * tag indicating the type of binding it supports. The entries MUST be formatted as follows:
+ * <p>
+ * <li>keys e.g. "vendorId:productId" -- each part comprising four <b>UPPER</b>-case hex characters.
+ * <li>entries e.g. "vendor OEM / sub-vendor name / ZigBee or Z-Wave" -- including a supported binding id.
+ * <p>
+ * Note: supported binding ids can be as follows:
+ * <li>'Zigbee'
+ * <li>'Z-Wave'
+ * <li>'EnOcean'
+ * Please do NOT include supported binding id in the case of very commonly used generic serial chips, since this may
+ * lead to false-positive discoveries.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+public class UsbProductDatabase {
+
+    // @formatter:off
+    public static final Map<String, String> PRODUCT_NAMES = Map.of(
```

Comment:
I've not looked through the code to see how everything is used, but I wonder if using `String` here for both the definition of the USB VID:PID and also the name of the device, and I guess the binding name is encoded here somewhere is the right approach? Wouldn't it be better to define a class that breaks this information out and also allows a descriptive name (eg Silicon labs) and the binding name (eg ZigBee) to be defined, and used, independently?

Reply:
Good idea. I could change that. However before reengineering such fine details, perhaps we need to consider the broader question of whether you guys even agree to the concept at all?

Reply:
But just also remember the other way round case too. The user has an OH system, he the buys a stick, plugs it in, and then looks for an addon to support it, and voila OH recommends the right binding for it.

Reply:
But don't get me wrong. I won't waste any more time developing something that nobody wants. I started the eork because I was told that people do want it.

Reply:
BTW I think there may be value in splitting this PR into two parts..
1. The USB device scanner for non Linux systems. Which may have limited functionality vs Linux. But more functionality than none at all. You may debate if that makes sense.
2. And add-on suggestion finder for addons that are related to specific USB devices (no matter how they may be discovered). You may debate if that adds value.

The advantage of that approach is that the two paths can be reviewed and progressed indepently. => WDYT?

Reply:
> I won't waste any more time developing something that nobody wants.

Again - this is just my view - others may think differently... To me, on balance it doesn't seem like it adds a lot over just searching the addons for zwave (or whatever). OH requires a lot of configuration, and to me this doesn't appreciably reduce that burden - but maybe for the new user there's still a use case there... Hopefully others will chime in with more thoughts :)

Reply:
> wonder if using String here for both the definition of the USB VID:PID and also the name of the device
> define a class that breaks this information out and also allows a descriptive name (eg Silicon labs) and the binding name (eg ZigBee) to be defined, and used, independently

=> continue discussion on these topics in #3930